cli/action.go

```diff
@@ -104,3 +106,29 @@ func action(c *cli.Context) error {
 
 	return nil
 }
+
+func gitignore(pattern, gitignorePath string) error {
+	f, err := os.OpenFile(gitignorePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return fmt.Errorf("open .gitignore file at %s: %s", gitignorePath, err)
+	}
+
+	contents, err := ioutil.ReadFile(gitignorePath)
+	matched, err := regexp.MatchString(fmt.Sprintf("(^%[1]s\n)|(\n%[1]s\n)", pattern), string(contents))
```

Comment:
(\n%[1]s\n) may not be needed as the first part mathces the beginning of the line.
Use $ instead of \n to match end of line.